azurerm/internal/services/apimanagement/api_management_api_resource.go

```diff
@@ -306,6 +353,24 @@ func resourceArmApiManagementApiCreateUpdate(d *schema.ResourceData, meta interf
 	subscriptionKeyParameterNamesRaw := d.Get("subscription_key_parameter_names").([]interface{})
 	subscriptionKeyParameterNames := expandApiManagementApiSubscriptionKeyParamNames(subscriptionKeyParameterNamesRaw)
 
+	authenticationSettings := &apimanagement.AuthenticationSettingsContract{}
+	if vs, hasOAuth2Authorization := d.GetOk("oauth2_authorization"); hasOAuth2Authorization {
+		oAuth2AuthorizationVs := vs.([]interface{})
+		oAuth2AuthorizationV := oAuth2AuthorizationVs[0].(map[string]interface{})
+		authenticationSettings.OAuth2 = &apimanagement.OAuth2AuthenticationSettingsContract{
+			AuthorizationServerID: utils.String(oAuth2AuthorizationV["authorization_server_name"].(string)),
+			Scope:                 utils.String(oAuth2AuthorizationV["scope"].(string)),
+		}
+	}
+	if vs, hasOpenIDAuthorization := d.GetOk("openid_authentication"); hasOpenIDAuthorization {
+		openIDAuthorizationVs := vs.([]interface{})
+		openIDAuthorizationV := openIDAuthorizationVs[0].(map[string]interface{})
```

Comment:
And same again here?

Reply:
Done in 778a38832e0127eaaf2fb5885eabc16ba90a3405.